api/shipper.proto

```diff
@@ -37,40 +37,45 @@ message PublishRequest {
 message Event {
  // Creation timestamp of the event.
  google.protobuf.Timestamp timestamp = 1;
- // Identifies the input that generated the event.
- Input input = 2;
+ // Source of the generated event.
+ Source source = 2;
  // Optional. Producers may specify a unique event ID if the event has a natural unique
  // identifier to simplify acknowledgement tracking. If no such identifier exists, the queue   
  // offset can be used for acknowledgement tracking.
- string event_id = 4;
+ string event_id = 3;
  // Data stream for the event.
- DataStream data_stream = 5;
+ DataStream data_stream = 4;
  // Metadata JSON object (map[string]google.protobuf.Value)
- google.protobuf.Struct metadata = 6;
+ google.protobuf.Struct metadata = 5;
  // Field JSON object (map[string]google.protobuf.Value)
- google.protobuf.Struct fields = 7;
+ google.protobuf.Struct fields = 6;
  // Note: The google.protobuf.Value type should be extended (or re-implemented) with a
  // google.protobuf.Timestamp to avoid the cost of encoding timestamps as strings and then 
  // parsing them back into a date representation.
 }
  
-// Input identifiers from the agent policy, see example configuration.
-message Input {
+// Source information required for proper event tracking, processing and routing
+message Source {
  // Input ID in the agent policy.
- string id = 1;
- // Input name in the agent policy.
- string name = 2;
- // Input type in the agent policy (e.g. logfile, winlog, endpoint, etc.)
- string type = 3;
+ // This value should be added to each produced event and forwarded via this proto field.
+ // When running under agent a processor is generated for adding this metadata.
```

Comment:
```suggestion
```

I would remove this extra context on beat implementation details. The top level comment on `Source` now explains why this field is needed.

Reply:
Okay, but I'm confused, you asked to add this in https://github.com/elastic/elastic-agent-shipper/pull/56#discussion_r897932862 but now we're removing it again.

Reply:
Yes, sorry I should only have made that comment on the streams ID.